src/prss.rs

```diff
@@ -0,0 +1,493 @@
+use crate::field::Field;
+use aes::{
+    cipher::{generic_array::GenericArray, BlockEncrypt, KeyInit},
+    Aes256,
+};
+use byteorder::{ByteOrder, NativeEndian};
+use hkdf::Hkdf;
+use rand::{CryptoRng, RngCore};
+use sha2::Sha256;
+use x25519_dalek::{EphemeralSecret, PublicKey};
+
```

Comment:
any paper or write-up you can link to describing the goal of the math being done here?

Reply:
I should do that write-up.  Let me get back to you on that though.